Crane/app/src/main/java/androidx/compose/samples/crane/home/LandingScreen.kt

```diff
@@ -32,7 +32,7 @@ private const val SplashWaitTime: Long = 2000
 @Composable
 fun LandingScreen(modifier: Modifier = Modifier, onTimeout: () -> Unit) {
     Box(modifier = modifier.fillMaxSize(), alignment = Alignment.Center) {
-        LaunchedTask {
+        LaunchedEffect(onTimeout) {
```

Comment:
Is this what we want? To restart the delay if the `onTimeout` action changes? It seemed like `rememberUpdatedState` would be preferable here?

Reply:
That's what @objcode suggested. If we use `rememberUpdatedState` we would also need `LaunchedEffect(Unit)` which makes `rememberUpdatedState` useless?

```
val currentOnTimeout by rememberUpdatedState(onTimeout)
LaunchedEffect(Unit) {
  ...
  currentOnTimeout() // this is equivalent to onTimeout() since this is never going to be called again on recomposition
}
```